ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureBlobLayoutStrategy.java

```diff
@@ -149,9 +155,13 @@ private String getAzureBlobName(CloudBlobMetadata blobMetadata) {
     String blobIdStr = blobMetadata.getId();
     int nameVersion = blobMetadata.getNameSchemeVersion();
     switch (nameVersion) {
-      default:
+      case 0:
         // Use the last four chars as prefix to assist in Azure sharding, since beginning of blobId has little variation.
         return blobIdStr.substring(blobIdStr.length() - 4) + BLOB_NAME_SEPARATOR + blobIdStr;
+      case 1:
+      default:
+        // snalli@: I have discussed thoroughly with MSFT that prefixing 4 random char does not help at all!
```

Comment:
I spoke with Claus https://www.linkedin.com/in/claus-joergensen-3128ba34/ for an hour and clarified with him over and over again beyond any doubt that this naming scheme of adding 3 random char has absolutely no benefit at all. You should bother him if you are still in doubt. Claus told me this document is stale and whoever implemented V1 ambry-backup probably didn't fully understand how Azure works. At the scale ambry is using azure, this naming scheme doesn't matter.

Hot partitions are taken care of by splitting the partition and Azure does this all the time - no matter what naming scheme we choose. And partitions are also merged. Partitions can be split if the txn rate exceeds a threshold or the size exceed a threshold (in terabytes) or the number of keys exceeds a threshold (in millions of keys). When the metrics go below the threshold, they are merged to save resources. During splits/merges, partitions may be unavailable and p99 of unavailability is 5 seconds, was his guess.

Can we _please_ move away from the decisions of V1 now and focus on some hard logical evidence ? 

More guidance from MSFT:

(HTBB doesn't apply to us. It is for azure's stitch requests and we don't use it)

If possible, to use blob or block sizes greater than 256 KiB for standard and premium storage accounts. Larger blob or block sizes automatically activate high-throughput block blobs. High-throughput block blobs provide high-performance ingest that isn't affected by partition naming. 
The duration of unavailability can vary from a few seconds to several minutes. Partitions can be split down as small as a blob family (the base blob and snapshots) or it may contain millions of blobs.  
We split, merge, and reassign partitions to balance load as needed and it happens all the time.  The naming guidance is really to try and avoid an "append only in keyspace" pattern of blob writes.  Splits make no difference (if you always append to the right side of a partition, splitting it in the middle does no good in terms of providing more resources and scaling out). 
If you take advantage of HTBB, naming almost doesn't make any difference.  It doesn't have to be hashed naming if the write pattern is naturally spread out (e.g. the blob names are person names and people's blobs get randomly updated).  
Using an exponential back off retry policy is recommended, and the client libraries default to this behavior. The exponential backoff allows the load on the partition to decrease, and to ease out spikes in traffic to that partition.